atomic_reactor/plugins/post_flatpak_create_oci.py

```diff
@@ -68,119 +34,63 @@ def __init__(self, workflow):
+        df_labels = build_dir.dockerfile_with_parent_env(
+            self.workflow.imageutil.base_image_inspect()
+        ).labels
 
-        df_labels = df_parser(self.workflow.df_path, workflow=self.workflow).labels
-        self.builder.add_labels(df_labels)
+        builder.add_labels(df_labels)
 
-        tarred_filesystem, manifest = self._export_filesystem()
-        self.log.info('filesystem tarfile written to %s', tarred_filesystem)
-        self.log.info('manifest written to %s', manifest)
+        tmp_dir = tempfile.mkdtemp(dir=build_dir.path)
 
-        image_components = self.builder.get_components(manifest)
-        self.workflow.data.image_components = image_components
+        image_filesystem = self.workflow.imageutil.extract_filesystem_layer(
+            str(build_dir.exported_squashed_image), str(tmp_dir))
 
-        ref_name, outfile, tarred_outfile = self.builder.build_container(tarred_filesystem)
+        filesystem_path = os.path.join(tmp_dir, image_filesystem)
 
-        # OSBS2 TBD
-        self.log.info('Marking filesystem image "%s" for removal', self.workflow.data.image_id)
+        with open(filesystem_path, 'rb') as f:
+            # this part is 'not ideal' but this function seems to be a prerequisite
+            # for building flatpak image since it does the setup for it
+            flatpak_filesystem, flatpak_manifest = builder._export_from_stream(f)
 
-        image_id = self._get_oci_image_id(outfile)
-        self.log.info('New OCI image ID is %s', image_id)
-        # OSBS2 TBD
-        self.workflow.data.image_id = image_id
+        self.log.info('filesystem tarfile written to %s', flatpak_filesystem)
 
-        labels = Labels(df_labels)
-        _, image_name = labels.get_name_and_value(Labels.LABEL_TYPE_NAME)
-        _, image_version = labels.get_name_and_value(Labels.LABEL_TYPE_VERSION)
-        _, image_release = labels.get_name_and_value(Labels.LABEL_TYPE_RELEASE)
+        image_rpm_components = builder.get_components(flatpak_manifest)
 
-        name = '{}-{}'.format(self.key, image_name)
-        tag = '{}-{}'.format(image_version, image_release)
-        # The OCI id is tracked by the builder. The image will be removed in the exit phase
-        # No need to mark it for removal after pushing to the local storage
-        self._copy_oci_to_local_storage(outfile, name, tag)
+        ref_name, outfile, _ = builder.build_container(flatpak_filesystem)
 
         metadata = get_exported_image_metadata(outfile, IMAGE_TYPE_OCI)
         metadata['ref_name'] = ref_name
 
         self.log.info('OCI image is available as %s', outfile)
 
-        metadata = get_exported_image_metadata(tarred_outfile, IMAGE_TYPE_OCI_TAR)
-        metadata['ref_name'] = ref_name
+        shutil.rmtree(tmp_dir)
+
+        return {'metadata': metadata, 'components': image_rpm_components}
+
+    def run(self) -> Optional[Dict[str, Dict[str, Union[str, int]]]]:
```

Comment:
I think the type should be `Optional[Dict[str, <ret. type of build_flatpak_image>]]`

Reply:
ah, that should be very helpful in future as well

Reply:
updated